impeller/docs/standalone_gles.md

```diff
@@ -0,0 +1,195 @@
+
+This guide details extremely low-level setup and the `//impeller/renderer` API directly above the HAL. Most users of Impeller will likely use the API using convenience wrappers already written for the platform. Interacting directly with the HAL is extremely powerful but also verbose. Applications are likely to also use higher level frameworks like Aiks or Display Lists.
+
+Building Impeller for the target platform is outside the scope of this guide.
+
+> [!CAUTION]
+> The code provided inline is pseudo-code and doesn't include error handling. See the headerdocs for more on error handling and failure modes. All classes are assumed to be in the `impeller` namespace. For a more complete example of setting up standalone Impeller, see [this patch](https://github.com/flutter/engine/pull/52472/files) that adds support for Impeller rendering via Wasm [in the browser and WebGL 2](https://public.chinmaygarde.com/impeller/wasm/wasm.html).
+
+# Setup
+
+To get started with Impeller rendering, you need to setup a context and a renderer. For backend specific classes, the convention in Impeller is to append the backend name as a suffix. So if you need to create an instance of an `impeller::Context` for OpenGLES, look for `impeller::ContextGLES`.
+
+In our case, we need a `impeller::ContextGLES` and `impeller::Renderer`.
+
+## Setting up WSI with the EGL Toolkit
+
+Before we get to `OpenGLES` we need to do a bit of WSI. While we can resort to using EGL directly, Impeller has a handy toolkit for it.
+
+First create an EGL display connection:
+
+```c++
+egl::Display display;
+```
+
+Ask the display for a valid EGL configuration. Impeller needs and OpenGL ES 2.0 configuration.
+
+```c++
+egl::ConfigDescriptor egl_desc;
+egl_desc.api = egl::API::kOpenGLES2;
+egl_desc.samples = egl::Samples::kOne;
+egl_desc.color_format = egl::ColorFormat::kRGBA8888;
+egl_desc.stencil_bits = egl::StencilBits::kZero;
+egl_desc.depth_bits = egl::DepthBits::kZero;
+egl_desc.surface_type = egl::SurfaceType::kWindow;
+
+auto config = display.ChooseConfig(egl_desc);
+```
+
+Once a valid config has been obtained, create a context and window surface. Creating the window surface requires a native window handle. Get the appropriate one for your platform. For instance, on Android, this is an `ANativeWindow`.
+
+```c++
+auto context = display.CreateContext(*config, nullptr /* sharegroup */ );
+auto surface = display.CreateWindowSurface(*config, native_window_handle);
+```
+
+Now that we have context, make it current on the calling thread. This should be fully setup WSI.
+
+```c++
+context->MakeCurrent(*surface);
+```
+
+## Creating the OpenGL ES Context
+
+Impeller doesn't statically link against OpenGL ES. You need to give it a callback the return the appropriate OpenGL ES function for given name. With EGL, this can be something as simple as:
```

Comment:
```suggestion
Impeller doesn't statically link against OpenGL ES. You need to give it a callback the returns the appropriate OpenGL ES function for given name. With EGL, this can be something as simple as:
```